Fix typing errors in torch.distributed.nn.* directory.

xuzhao9 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #47534 Fix typing errors in torch.distributed.*, close issue #42967.
* **#47533 Fix typing errors in torch.distributed.nn.* directory.**
* #47532 Fix typing errors in torch.distributed.distributed_c10d.*
* #47531 Fix type annotation errors in torch.distributed.* directory

Differential Revision: [D24952500](https://our.internmc.facebook.com/intern/diff/D24952500)

xuzhao9 committed:
Fix typing errors in torch.distributed.nn.* directory.

[ghstack-poisoned]

xuzhao9 commented on torch/distributed/nn/jit/instantiator.py:
```diff
@@ -37,11 +38,12 @@ def get_arg_return_types_from_interface(module_interface):
 
         if argument.has_default_value():
-            default_value_str = " = {}".format(argument.default)
+            default_value_str = " = {}".format(argument.default)  # type: ignore[attr-defined]
```
IMO Argument.default_value should be used here.

malfet replied:
Can you please change it to `default_value` rather than suppressing an error here?
```suggestion
            default_value_str = " = {}".format(argument.default_value)
```

malfet commented on torch/distributed/nn/api/remote_module.py:
```diff
@@ -210,109 +211,109 @@ def remote_parameters(self, recurse: bool = True) -> List[rpc.RRef[Parameter]]:
         _raise_not_supported(self.register_parameter.__name__)
 
-    def add_module(self, name: str, module: Optional["Module"]) -> None:
+    def add_module(self, name: str, module: Optional["Module"]) -> None:  # type: ignore[return]
```
Since you are importing module now, you don't need to escape it in quotes, do you?
```suggestion
    def add_module(self, name: str, module: Optional[Module]) -> None:  # type: ignore[return]
```

malfet commented on torch/distributed/nn/api/remote_module.py:
```diff
@@ -210,109 +211,109 @@ def remote_parameters(self, recurse: bool = True) -> List[rpc.RRef[Parameter]]:
         _raise_not_supported(self.bfloat16.__name__)
 
-    def to(self, *args, **kwargs):
+    def to(self, *args, **kwargs):  # type: ignore[return]
```
Aren't `to` returning `T`?
```suggestion
    def to(self: T, *args, **kwargs) -> T:  # type: ignore[return]
```

walterddr commented on torch/distributed/nn/api/remote_module.py:
```diff
@@ -210,109 +211,109 @@ def remote_parameters(self, recurse: bool = True) -> List[rpc.RRef[Parameter]]:
         _raise_not_supported(self.to.__name__)
 
-    def register_backward_hook(
+    def register_backward_hook(  # type: ignore[return]
         self, hook: Callable[["Module", _grad_t, _grad_t], Union[None, Tensor]]
     ) -> RemovableHandle:
```
should we swap the order of `:`?
```suggestion
    def register_backward_hook(  # type: ignore[return]
        self, hook: Callable[["Module", _grad_t, _grad_t], Union[None, Tensor]]
    ) -> RemovableHandle:  # type: ignore[return]
```

walterddr commented on torch/distributed/nn/api/remote_module.py:
```diff
@@ -210,109 +211,109 @@ def remote_parameters(self, recurse: bool = True) -> List[rpc.RRef[Parameter]]:
         )
 
-    def named_parameters(
+    def named_parameters(  # type: ignore[return]
```
multiline type ignore is fixed a while back. we can put all these at the end of a AST scope

xuzhao9 replied:
Sorry I can't reproduce it locally (after rebasing and recompiling on master). Do I also need to update the mypy version with conda?

xuzhao9 replied:
Moving the type ignore to the end of a multiline statement actually fails the test: https://app.circleci.com/pipelines/github/pytorch/pytorch/237185/workflows/9d10924c-84e9-471b-a9ae-ffe3d6524a7a/jobs/8817770. Is it because in circleci we are using an old version of mypy?

xuzhao9 replied:
This feature is supported by only Python > 3.8, so we don't support it here.

walterddr commented on mypy.ini:
```diff
@@ -62,6 +62,9 @@ ignore_errors = False
 [mypy-torch.distributed.distributed_c10d.*]
 ignore_errors = False
 
+[mypy-torch.distributed.nn.*]
+ignore_errors = False
```
I am not 100% sure how this usually work on ghstack - do you have to wait for all 4 PRs to be accepted and commit in order? (seems like if all PR merged together you can just remove this?)

xuzhao9 replied:
Yes. After all 4 PRs merged together, this line (and all other torch.distributed.* related lines) will be removed.

xuzhao9 committed:
Update on "Fix typing errors in torch.distributed.nn.* directory."




[ghstack-poisoned]

xuzhao9 committed:
Update on "Fix typing errors in torch.distributed.nn.* directory."




[ghstack-poisoned]

xuzhao9 committed:
Update on "Fix typing errors in torch.distributed.nn.* directory."




[ghstack-poisoned]

xuzhao9 committed:
Update on "Fix typing errors in torch.distributed.nn.* directory."




[ghstack-poisoned]

xuzhao9 committed:
Update on "Fix typing errors in torch.distributed.nn.* directory."




[ghstack-poisoned]

xuzhao9 committed:
Update on "Fix typing errors in torch.distributed.nn.* directory."




[ghstack-poisoned]

xuzhao9 committed:
Update on "Fix typing errors in torch.distributed.nn.* directory."




[ghstack-poisoned]

walterddr reviewed (approved):
lgtm

xuzhao9 committed:
Update on "Fix typing errors in torch.distributed.nn.* directory."


Differential Revision: [D24952500](https://our.internmc.facebook.com/intern/diff/D24952500)

[ghstack-poisoned]

xuzhao9 committed:
Update on "Fix typing errors in torch.distributed.nn.* directory."


Differential Revision: [D24952500](https://our.internmc.facebook.com/intern/diff/D24952500)

[ghstack-poisoned]

xuzhao9 committed:
Update on "Fix typing errors in torch.distributed.nn.* directory."


Differential Revision: [D24952500](https://our.internmc.facebook.com/intern/diff/D24952500)

[ghstack-poisoned]

xuzhao9 committed:
Update on "Fix typing errors in torch.distributed.nn.* directory."


Differential Revision: [D24952500](https://our.internmc.facebook.com/intern/diff/D24952500)

[ghstack-poisoned]

xuzhao9 committed:
Update on "Fix typing errors in torch.distributed.nn.* directory."


Differential Revision: [D24952500](https://our.internmc.facebook.com/intern/diff/D24952500)

[ghstack-poisoned]
